disnake/guild_scheduled_event.py

```diff
@@ -348,6 +349,23 @@ async def edit(
     ) -> GuildScheduledEvent:
         ...
 
+    # new channel, no entity_type
+    @overload
+    async def edit(
+        self,
+        *,
+        channel: Optional[Snowflake],
+        name: str = ...,
+        description: Optional[str] = ...,
+        image: Optional[AssetBytes] = ...,
+        privacy_level: GuildScheduledEventPrivacyLevel = ...,
+        scheduled_start_time: datetime = ...,
+        scheduled_end_time: Optional[datetime] = ...,
+        status: GuildScheduledEventStatus = ...,
+        reason: Optional[str] = ...,
+    ) -> GuildScheduledEvent:
+        ...
```

Comment:
To cover all cases, I believe we need an additional overload here - essentially, the end goal for these overloads is:
1. generic, no type change (should accept all parameters)
2. `entity_type=external` specified, `channel=None` may be provided but doesn't have to be
3. `entity_type=voice/stage` specified, `channel=<valid>` may be provided but doesn't have to be
4. `channel=None` specified
5. `channel=<valid>` specified

(I hope that makes sense, and that I didn't miss anything)

A `channel: None = ...,` is also needed in overload no.2 (can't suggest it directly because GitHub) to match the above description and allow calls like
```py
await e.edit(
    entity_type=disnake.GuildScheduledEventEntityType.external,
    channel=None,
    entity_metadata=disnake.GuildScheduledEventMetadata(location="h"),
)
```

These would be (4) and (5) in the list above:
```suggestion
    # channel=None, no entity_type
    @overload
    async def edit(
        self,
        *,
        channel: None,
        name: str = ...,
        description: Optional[str] = ...,
        image: Optional[AssetBytes] = ...,
        privacy_level: GuildScheduledEventPrivacyLevel = ...,
        scheduled_start_time: datetime = ...,
        scheduled_end_time: datetime = ...,
        entity_metadata: GuildScheduledEventMetadata = ...,
        status: GuildScheduledEventStatus = ...,
        reason: Optional[str] = ...,
    ) -> GuildScheduledEvent:
        ...

    # valid channel, no entity_type
    @overload
    async def edit(
        self,
        *,
        channel: Snowflake,
        name: str = ...,
        description: Optional[str] = ...,
        image: Optional[AssetBytes] = ...,
        privacy_level: GuildScheduledEventPrivacyLevel = ...,
        scheduled_start_time: datetime = ...,
        scheduled_end_time: Optional[datetime] = ...,
        status: GuildScheduledEventStatus = ...,
        reason: Optional[str] = ...,
    ) -> GuildScheduledEvent:
        ...
```